javalib/src/main/scala/java/net/HttpCookie.scala

```diff
@@ -0,0 +1,413 @@
+
+      var i = 0
+      while (i < unprocessed.size()) {
+        val kv = unprocessed.get(i)
+        val eqIndex = kv.indexOf("=")
+        val name = kv.substring(0, eqIndex)
+        val value = kv.substring(eqIndex + 1)
+
+        val cookie = parseValue(name.trim(), unquote(value.trim()))
+        if (cookie != null) {
+          cookies.add(cookie)
+        }
+
+        i += 1
+
+      }
+
+    }
+
+    cookies
+
+  }
+
+  // Splits the string at the given character,
+  // however it also respects quoted strings, i.e.
+  // occurences of the separator inside quoted strings
+  // won't trigger a split
+  // backslash also escapes double quotes inside the quoted sections
+  private def splitUnquotedAt(
+      in: String,
+      separator: Char
+  ): java.util.List[String] = {
+
+    val res = new ArrayList[String]()
+    if (in == "") {
+      res
+    } else {
+
+      var i = 0
+      var begin = 0
+      var inQuotes = false
+      var escaping = false
+
+      while (i < in.length()) {
+
+        val char = in.charAt(i)
+
+        if (char == separator && !inQuotes) {
+          res.add(in.substring(begin, i))
+          begin = i + 1
+        } else if (char == '\\' && inQuotes && !escaping) {
+          escaping = true
+        } else if (char == '"' && inQuotes && !escaping) {
+          inQuotes = false
+        } else if (char == '"' && !inQuotes) {
+          inQuotes = true
+        }
+
+        i += 1
+
+      }
+
+      res.add(in.substring(begin))
+
+      res
+    }
+  }
+
+  /** If the string begins and ends with quotes - remove them, otherwise return
+   *  the input
+   */
+  private def unquote(in: String): String = {
+    if (in.charAt(0) == '"' && in.charAt(in.length() - 1) == '"') {
+      in.substring(1, in.length() - 1)
+    } else in
+  }
+
+  /** performs parsing on the section of the cookie following the cookie name,
+   *  and constructs the resulting cookie. If the given value contains multiple
+   *  cookies - this method will break, so any input into this method must be
+   *  pre-processed such that only the section that pertains to a single cookie
+   *  is fed into this method
+   */
+  private def parseValue(name: String, value: String): HttpCookie = {
+
+    val attrs = splitUnquotedAt(value, ';')
+
+    if (attrs.size() == 0) {
+      new HttpCookie(name, "")
+    } else {
+
+      val value = attrs.get(0)
+
+      val cookie = new HttpCookie(name, unquote(value.trim()))
+
+      var i = 1
+      while (i < attrs.size()) {
+
+        val prop = attrs.get(i).trim()
+
+        val sep = prop.indexOf('=')
+
+        val propValue: String = if (sep == -1) {
+          null
+        } else {
+          unquote(prop.substring(sep + 1).trim())
+        }
+
+        val propName =
+          if (sep == -1)
+            prop
+          else {
+            prop.substring(0, sep).trim()
+          }
+
+        if (propName.regionMatches(true, 0, Discard, 0, Discard.length())) {
```

Comment:
How about `equalsIgnoreCase` instead? Again, you're comparing the whole string. Alternatively you can convert the `prop.toLowerCase` and then make a simple match - would be way more readable.